tests/components/eafm/test_sensor.py

```diff
@@ -0,0 +1,431 @@
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 56},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        },
+    )
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "56"
+
+
+async def test_reading_is_sampled(hass, mock_get_station):
+    """Test that a sensor is added and polled."""
+    await async_setup_test_fixture(
+        hass,
+        mock_get_station,
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        },
+    )
+
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+    assert state.attributes["unit_of_measurement"] == "m"
+
+
+async def test_multiple_readings_are_sampled(hass, mock_get_station):
+    """Test that multiple sensors are added and polled."""
+    await async_setup_test_fixture(
+        hass,
+        mock_get_station,
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                },
+                {
+                    "@id": "really-long-unique-id-2",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Second Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 4},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                },
+            ],
+        },
+    )
+
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+    assert state.attributes["unit_of_measurement"] == "m"
+
+    state = hass.states.get("sensor.my_station_water_level_second_stage")
+    assert state.state == "4"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)